gui.py

```diff
@@ -87,37 +83,63 @@ def __init__(self, matrix, data_a, data_b, root):
         super().__init__(root)
         self.pack(fill=tk.BOTH, expand="true")
         self._zoom_map = ZoomMap(matrix)
-        self._map = tk.Label(self, image=self._zoom_map.image)
-        self._map.pack()
+        self._map = _Map(self, self._zoom_map)
+
+        self._contexts = [_Context(self, data, self._zoom_map)
+                          for data in (data_a, data_b)]
+        [self._map.bind(event, self._on_motion)
+                for event in ["<Motion>", "<Enter>"]]
 
+    def _on_motion(self, event):
         # We're using (row, col) format, so the first one changes with Y.
-        self._contexts = [_Context(self, data, axis, self._zoom_map)
-                          for data, axis in ((data_a, "y"), (data_b, "x"))]
-        [self._map.bind(*args) for args in
-                [("<Button-4>",  functools.partial(self._zoom, -1)),
-                 ("<Button-5>",  functools.partial(self._zoom,  1)),
+        self._contexts[0].display(self._map.canvasy(event.y))
+        self._contexts[1].display(self._map.canvasx(event.x))
+
+
+class _Map(tk.Canvas):
+    def __init__(self, tk_parent, zoom_map):
+        # TODO: figure out a way to prevent panning off the map, so we never
+        # see the green background.
+        # TODO: Modify this so we only have part of the map stored in an image
+        # at a time, so we can display really big programs without running out
+        # of memory.
+        super().__init__(tk_parent, height=500, width=500, bg="green",
+                         xscrollincrement=1, yscrollincrement=1)
+        self._zoom_map = zoom_map
+        self._set_image()
+        self.pack()
+        [self.bind(*args) for args in
+                [("<Button-4>",  partial(self._zoom, -1)),
+                 ("<Button-5>",  partial(self._zoom,  1)),
                  ("<Button-1>",  self._on_click),
-                 ("<B1-Motion>", self._on_drag),
-                 ("<Motion>",    self._on_motion),
-                 ("<Enter>",     self._on_motion)]]
+                 ("<B1-Motion>", self._on_drag)]]
 
-    def _on_motion(self, event):
-        [context.display(event) for context in self._contexts]
+    def _set_image(self):
+        self._image = self.create_image(0, 0, anchor=tk.NW,
+                                        image=self._zoom_map.image)
 
     def _zoom(self, amount, event):
-        # TODO: when click-and-drag is implemented and the whole map is not
-        # onscreen, make the zooming centered around the curser.
-        self._zoom_map.zoom(amount)
-        self._map.configure(image=self._zoom_map.image)
+        did_zoom = self._zoom_map.zoom(amount)
+        if not did_zoom:
+            return
```

Comment:
Yeah, you might be right. I named the variable at first because it wasn't obvious what `.zoom()` should return. but if it's in an if statement, it likely returns a boolean, and there's only one boolean I could imagine it returning, so we don't need to name it after all.